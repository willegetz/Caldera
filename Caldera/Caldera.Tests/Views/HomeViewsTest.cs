using System.Web.Mvc;
using Caldera.Controllers;
using NUnit.Framework;
using ApprovalTests.Asp.Mvc;
using System;
using ApprovalTests.Asp;
using ApprovalTests.Reporters;
using Caldera.Tests.TestableControllers;
using System.Text.RegularExpressions;

namespace Caldera.Tests.Views
{
    [TestFixture]
    [UseReporter(typeof(BeyondCompareReporter), typeof(FileLauncherReporter), typeof(ClipboardReporter))]
    public class HomeViewsTest
    {
        // TODO: Start Without Debugging to launch IISExpress and be able to use the MvcApprovals.
        [Test]
        public void IndexView()
        {
            PortFactory.MvcPort = 49264;
            MvcApprovals.VerifyMvcPage<TestableHomeController>(home => home.TestIndex, CopyRightScrubber);
        }

        public static string CopyRightScrubber(string input)
        {
            string copyrightTag = @"<p>&copy; \d{4}";
            return Regex.Replace(input, copyrightTag, "<p>&copy; 0000");
        }

        [Test]
        public void AboutView()
        {
            PortFactory.MvcPort = 49264;
            MvcApprovals.VerifyMvcPage<TestableHomeController>(home => home.TestAbout, CopyRightScrubber);
        }

        [Test]
        public void ContactView()
        {
            PortFactory.MvcPort = 49264;
            MvcApprovals.VerifyMvcPage<TestableHomeController>(home => home.TestContact, CopyRightScrubber);
        }
    }
}
