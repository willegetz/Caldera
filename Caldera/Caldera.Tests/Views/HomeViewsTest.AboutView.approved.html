<!DOCTYPE html>
<html>
<head><base href="http://localhost:49264/TestableHome">
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>About - My ASP.NET Application</title>
    <link href="/Content/bootstrap.css" rel="stylesheet"/>
<link href="/Content/site.css" rel="stylesheet"/>

    <script src="/Scripts/modernizr-2.6.2.js"></script>


</head>
<body>
    <div class="navbar navbar-inverse navbar-fixed-top">
        <div class="container">
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="/">Application name</a>
            </div>
            <div class="navbar-collapse collapse">
                <ul class="nav navbar-nav">
                    <li><a href="/">Home</a></li>
                    <li><a href="/Home/About">About</a></li>
                    <li><a href="/Home/Contact">Contact</a></li>
                </ul>
                    <ul class="nav navbar-nav navbar-right">
        <li><a href="/Account/Register" id="registerLink">Register</a></li>
        <li><a href="/Account/Login" id="loginLink">Log in</a></li>
    </ul>

            </div>
        </div>
    </div>
    <div class="container body-content">
        

<h2>About.</h2>
<h3>Your application description page.</h3>

<p>Use this area to provide additional information.</p>

        <hr />
        <footer>
            <p>&copy; 0000 - My ASP.NET Application</p>
        </footer>
    </div>

    <script src="/Scripts/jquery-1.10.2.js"></script>

    <script src="/Scripts/bootstrap.js"></script>
<script src="/Scripts/respond.js"></script>

    
</body>
</html>
